src/main/java/nextstep/subway/line/domain/LineStationSorter.java

```diff
@@ -0,0 +1,54 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class LineStationSorter {
+
+    private static final int FIRST_INDEX = 0;
+
+    public static List<Station> sort(List<Section> sections) {
+        List<Station> stations = new ArrayList<>();
+        Station currentStation = findFirstStation(sections);
+
+        while(currentStation != null) {
+            stations.add(currentStation);
+            currentStation = findNextStation(currentStation, sections);
+        }
+
+        return stations;
+    }
+
+    private static Station findNextStation(Station currentStation, List<Section> sections) {
+        Station beforeDownStation = currentStation;
+
+        Optional<Section> nextSection = sections.stream()
+                .filter(section -> section.getUpStation().equals(beforeDownStation))
+                .findFirst();
+
+        return nextSection.isPresent() ? nextSection.get().getDownStation() : null;
```

Comment:
```suggestion
        return nextSection.map(Section::getDownStation).orElse(null);
```
이렇게 쓸수도 있겠네요! 😀
그리고 개인적으로 null을 리턴하는것보다 Optional을 리턴하는게 낫지 않나 생각해요.